aspnetcore/host-and-deploy/proxy-load-balancer.md

```diff
@@ -0,0 +1,134 @@
+
+The Forwarded Headers Middleware, from the [Microsoft.AspNetCore.HttpOverrides](https://www.nuget.org/packages/Microsoft.AspNetCore.HttpOverrides/) package, reads these headers and fills in the associated fields on [HttpContext](/dotnet/api/microsoft.aspnetcore.http.httpcontext). 
+
+The middleware updates:
+
+* [HttpContext.Connection.RemoteIpAddress](/dotnet/api/microsoft.aspnetcore.http.connectioninfo.remoteipaddress) &ndash; Set using the `X-Forwarded-For` header value. Additional settings influence how the middleware sets `RemoteIpAddress`. For details, see the [Forwarded Headers Middleware options](#forwarded-headers-middleware-options).
+* [HttpContext.Request.Scheme](/dotnet/api/microsoft.aspnetcore.http.httprequest.scheme) &ndash; Set using the `X-Forwarded-Proto` header value.
+
+When using any type of authentication middleware, the Forwarded Headers Middleware must run first. This ordering ensures that the authentication middleware can consume the header values and generate correct redirect URIs.
+
+Note that not all network appliances add the `X-Forwarded-For` and `X-Forwarded-Proto` headers without additional configuration. Consult your appliance manufacturer's guidance if the proxied requests don't contain these headers when they reach the app.
+
+Forwarded Headers Middleware [default settings](#forwarded-headers-middleware-options) can be configured. The default settings are:
+
+* There is only *one proxy* between the app and the source of the requests.
+* Only loopback addresses are configured for known proxies and known networks.
+
+## IIS and ASP.NET Core Module
+
+Forwarded Headers Middleware is enabled by default by IIS Integration Middleware when the app is run behind IIS and the ASP.NET Core Module. Forwarded Headers Middleware is activated with a restricted configuration specific to the ASP.NET Core Module due to trust concerns with forwarded headers (for example, [IP spoofing](https://www.iplocation.net/ip-spoofing)). The middleware is configured to forward the `X-Forwarded-For` and `X-Forwarded-Proto` headers. If additional configuration is required, see the [Forwarded Headers Middleware options](#forwarded-headers-middleware-options).
+
+## Other proxy server and load balancer scenarios
+
+Forwarded Headers Middleware isn't enabled by default, it must be enabled for an app to process forwarded headers with [UseForwardedHeaders](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersextensions.useforwardedheaders). After enabling the middleware if no [ForwardedHeadersOptions](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions) are specified to the middleware, the default [ForwardedHeadersOptions.ForwardedHeaders](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.forwardedheaders) are [ForwardedHeaders.None](/dotnet/api/microsoft.aspnetcore.httpoverrides.forwardedheaders).
```

Comment:
isn't enabled by default outside of IIS...

Reply:
Link to our Apache and NGinx docs. We'll need to update the config in those docs to include fowarders. E.g. https://www.nginx.com/resources/wiki/start/topics/examples/forwarded/ and `$proxy_add_x_forwarded_for` and https://www.digitalocean.com/community/tutorials/how-to-set-up-nginx-load-balancing-with-ssl-termination for x-forwarded-for.

Apache:
https://community.rackspace.com/products/f/public-cloud-forum/921/configuring-apache-for-ssl-termination-on-a-load-balancer
https://httpd.apache.org/docs/2.4/mod/mod_proxy.html (x-forwarded-for is implicit)

Reply:
> isn't enabled by default outside of IIS...

Yeah ... this goes back to my earlier point. I wanted "Other proxy server and load balancer scenarios" to make it clear. I'll add that tho because it makes it even more clear.

RE: The updates to the other docs. I have another PR touching one of those. I think it will break trying to do both at the same time. I *can* deal with a merge conflict, but I'd prefer to let that other one merge first (and this one) and then deal with this update. How about I follow this (and the other PR) up immediately with another PR to fix them?

Reply:
However you need to juggle it. Add notes to the issue for anything not being addressed in the current PR.

Reply:
Will do. I'd just like to avoid a giant :boom: if possible. Right now, this looks like a big 'ole :bomb:. I'll open an issue as soon as I'm done addressing your feedback.